packages/components/src/common/locale/index.js

```
@@ -69,16 +67,23 @@ export function getLangFromLocale(locale) {
 }
 
 /**
- * Provides corresponding country code (iso2) for provided locale
- * if the value is invalid returns null
+ * Provides corresponding country code (iso2) for locales code with explicit region value (`es-ES`, `en-GB`, `ja-JP` etc.)
+ * if the value is invalid or missing region it returns null
  *
- * @param locale (`es`, `es-ES`, `en-GB`, `en`, `ja`, `ja-JP` etc)
+ * @param locale
  * @returns {string|null}
  */
 export const getCountryFromLocale = (locale) => {
   const adjustedLocale = adjustLocale(locale);
   if (adjustedLocale === null) {
     return null;
   }
-  return adjustedLocale.slice(0, COUNTRY_ISO2_CODE_LENGTH);
+  try {
+    const { region } = new Intl.Locale(adjustedLocale);
+    return region ?? null;
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(error);
```

I don't think there is any benefit in logging an error here, do you?

---

it's considered as a bad practice to ignore errors at catch statement, it could be useful for avoiding/discovering bugs